### PR TITLE
Bump solana_rbpf, tower-lsp, and contract-metadata crates

### PR DESCRIPTION
**Pull request by seanyoung**
Signed-off-by: Sean Young <sean@mess.org>

### CODE REVIEW

**Review by xermicus — APPROVED** [review]
LGTM

File: solang-parser/Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ itertools = "0.10"
 serde = { version = "1.0", features = ["derive"], optional = true }
 
 [dev-dependencies]
-walkdir = "2.3.2"
+walkdir = "2.3"
```
[the remark this answers is not in the document]
Comment (LucasSte) [reply solang-parser/Cargo.toml:27]: > So this makes the dependency consistent with all the other entries in `Cargo.toml`.

All other entries in `slang-parser/Cargo.toml` follow this setting. Is our wish not to specify patch for the top-level `Cargo.toml` file too?

[the remark this answers is not in the document]
Comment (seanyoung) [reply solang-parser/Cargo.toml:27]: I guess so. The more I think about it, the more I think this really does not matter.

**Review by LucasSte — APPROVED** [review]

File: solang-parser/Cargo.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (LucasSte) [reply solang-parser/Cargo.toml]: Looks good.

**Pull request merged**